src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -246,11 +274,23 @@ public static Complex Sinh(Complex value)
 
         public static Complex Asin(Complex value)
         {
-            if ((value._imaginary == 0 && value._real < 0) || value._imaginary > 0)
+            double b, bPrime, v;
+            Asin_Internal(Math.Abs(value.Real), Math.Abs(value.Imaginary), out b, out bPrime, out v);
```

Comment:
Which version of the C# compiler are we on (I can't remember)? If we are on 2.0, then we should probably make use of `out var`. This function is also a good candidate to be implemented as a `local function`.